reducer/ingest/agent_span.cc

```diff
@@ -536,7 +536,7 @@ void AgentSpan::public_to_private_ipv4(
 
   auto &addr_map = global_private_to_public_address_map();
 
-  if (auto existing_public = addr_map.get(private_addr); existing_public) {
+  if (auto existing_public = addr_map.get(private_addr); existing_public && *existing_public == public_addr) {
```

Comment:
Shouldn't the warning be printed if the new public address differs from the exiting, and suppressed if it is the same?